Src/VimCore/CoreTypes.fs

```diff
@@ -52,6 +52,81 @@ type JoinKind =
     | RemoveEmptySpaces
     | KeepEmptySpaces
 
+[<StructuralEquality>]
+[<NoComparison>]
+[<Struct>]
+[<DebuggerDisplay("{ToString(),nq}")>]
+type SelectedSpan =

```

Comment:
The Visual Studio type is indeed just called `Microsoft.VisualStudio.Text.Selection` but reusing that name would lead to at least three types with the same name.

I chose "selected span" to mean "a span when it is selected" rather than "a span that is currently selected".

The reason I like `SelectedSpan` is so that the two APIs `GetSelectedSpans` and `SetSelectedSpans` include the type of the expected arguments. But I could be persuaded to rename it `SelectionSpan` like this:

```
    /// Get all the selected spans for the specified text view
    abstract GetSelectedSpans: textView: ITextView -> SelectionSpan seq

    /// Set all the selected spans for the specified text view
    abstract SetSelectedSpans: textView: ITextView -> SelectionSpan seq -> unit
```

Also, I see now that I need to add documentation comments to the entire type and its members.  Oops!